shuup/admin/modules/media/views.py

```diff
@@ -210,36 +275,61 @@ def handle_upload(self):
         })
 
     def handle_post_rename_folder(self, data):
-        folder = Folder.objects.get(pk=data["id"])
+        shop = get_shop(self.request)
+        folder = _get_folder_query(shop).get(pk=data["id"])
+        if _is_folder_shared(folder):
+            message = _("Can not rename shared folder.")
+            return JsonResponse({"success": False, "message": message})
+
         folder.name = data["name"]
         folder.save(update_fields=("name",))
         return JsonResponse({"success": True, "message": _("Folder renamed.")})
 
     def handle_post_delete_folder(self, data):
-        folder = Folder.objects.get(pk=data["id"])
+        shop = get_shop(self.request)
+        folder = _get_folder_query(shop).get(pk=data["id"])
+        if _is_folder_shared(folder):
+            message = _("Can not delete shared folder.")
+            return JsonResponse({"success": False, "message": message})
+
         new_selected_folder_id = folder.parent_id
         message = delete_folder(folder)
         return JsonResponse({"success": True, "message": message, "newFolderId": new_selected_folder_id})
 
     def handle_post_rename_file(self, data):
-        file = File.objects.get(pk=data["id"])
+        shop = get_shop(self.request)
+        file = _get_file_query(shop).get(pk=data["id"])
+        if _is_file_shared(file):
+            message = _("Can not rename shared file.")
+            return JsonResponse({"success": False, "message": message})
+
         file.name = data["name"]
         file.save(update_fields=("name",))
         return JsonResponse({"success": True, "message": _("File renamed.")})
 
     def handle_post_delete_file(self, data):
-        file = File.objects.get(pk=data["id"])
+        shop = get_shop(self.request)
+        file = _get_file_query(shop).get(pk=data["id"])
+        if _is_file_shared(file):
+            message = _("Can not delete shared file.")
+            return JsonResponse({"success": False, "message": message})
+
         try:
             file.delete()
         except IntegrityError as ie:
             raise Problem(str(ie))
         return JsonResponse({"success": True, "message": _("File deleted.")})
 
     def handle_post_move_file(self, data):
-        file = File.objects.get(pk=data["file_id"])
+        shop = get_shop(self.request)
+        file = _get_file_query(shop).get(pk=data["file_id"])
```

Comment:
handle `DoesNotExist` ?

Reply:
If these untranslated "not found" errors starts to happen then let's add the translation separately. These are catched at post already.